tests/acl/test_acl.py

```diff
@@ -106,34 +106,46 @@ def setup(duthosts, rand_one_dut_hostname, tbinfo, ptfadapter):
     mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
 
     # Get the list of upstream/downstream ports
-    downstream_ports = []
-    upstream_ports = []
+    downstream_ports = defaultdict(list)
+    upstream_ports =  defaultdict(list)
     downstream_port_ids = []
     upstream_port_ids = []
 
     topo = tbinfo["topo"]["type"]
     for interface, neighbor in mg_facts["minigraph_neighbors"].items():
         port_id = mg_facts["minigraph_ptf_indices"][interface]
         if (topo == "t1" and "T0" in neighbor["name"]) or (topo == "t0" and "Server" in neighbor["name"]):
-            downstream_ports.append(interface)
+            downstream_ports[neighbor['namespace']].append(interface)
             downstream_port_ids.append(port_id)
         elif (topo == "t1" and "T2" in neighbor["name"]) or (topo == "t0" and "T1" in neighbor["name"]):
-            upstream_ports.append(interface)
+            upstream_ports[neighbor['namespace']].append(interface)
             upstream_port_ids.append(port_id)
 
     # Get the list of LAGs
     port_channels = mg_facts["minigraph_portchannels"]
 
     # TODO: We should make this more robust (i.e. bind all active front-panel ports)
-    acl_table_ports = []
+    acl_table_ports =  defaultdict(list)
 
     if topo == "t0" or tbinfo["topo"]["name"] in ("t1", "t1-lag"):
-        acl_table_ports += downstream_ports
+        for namespace, port in downstream_ports.iteritems():
+            acl_table_ports[namespace] += port
+            # In multi-asic we need config both in host and namespace.
+            if namespace:
+                acl_table_ports[''] += port
 
     if topo == "t0" or tbinfo["topo"]["name"] in ("t1-lag", "t1-64-lag", "t1-64-lag-clet"):
-        acl_table_ports += port_channels
+        for k, v in port_channels.iteritems():
+            acl_table_ports[v['namespace']].append(k)
+            # In multi-asic we need config both in host and namespace.
+            if v['namespace']:
+                acl_table_ports[''].append(k)
     else:
-        acl_table_ports += upstream_ports
+        for namespace, port in upstream_ports.iteritems():
+            acl_table_ports[namespace] += port
+            # In multi-asic we need config both in host and namespace.
+            if namespace:
+                acl_table_ports[''] += port
```

Comment:
Why use an empty string as dict key? Can you use something more meaningful as dict key?

Reply:
Empty string namespace represents Default/host namespace. 
sonic-buildimage represents it with empty string https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-py-common/sonic_py_common/multi_asic.py#L26

However in sonic-mgmt we have made that macro as None. Changing it to empty string will have implication as lot of changes have evolved around it.